cql3/expr/expression.hh

```diff
@@ -184,6 +194,9 @@ const binary_operator* find_atom(const expression& e, Fn f) {
                 }
                 return nullptr;
             },
+            [] (const column_value&) -> const binary_operator* { return nullptr; },
```

Comment:
Well, atom doesn't describe binary_operator either. Atoms (the greek ones at least) were supposed to be indivisible.

btw, the patch doesn't change find_atom's return value - it can only return binary_operators.

Perhaps it should be renamed to

```c++
template <subexpression Subexpression, std::regular_invocable<const S&> Func>
const Subexpression*
find_subexpression(const expression& e, Func func);
```

(where `subexpression` is a concept that is true for any type that can be an `expression`)

Which will allow it to find anything.

Reply:
I like that renaming; it can be a follow-up.